### ISSUE TITLE
Firebase SDK for VisionOS in Unity package

### ISSUE THREAD

**Issue by Rusus24** [firebase/firebase-unity-sdk#933]
### Description

Hey!

Now we're trying to port a game from Oculus Quest 2 to Apple Vision headset. We developed the app with Unity3D. The Oculus runs on android and works great with firebase but after research it seems that the Unity Firebase package doesn't support VisionOS.

Is this feature taking into account in the near future?

### Firebase Product(s)

Authentication, Database, Storage

**Comment by maxcelar** [comment]
We also need Unity Vision OS support as soon as possible.

**Comment by skoteskote** [comment]
Same. This is currently blocking us from porting our project to VisionOS, so we hope it will be implemented as soon as possible.

**Comment by m-brewster** [comment]
Same, this is incredibly crucial to our project and blocking our ability to port over to VisionOS.

**Comment by a-maurice** [comment]
We are looking into it, though it will take some time, as some of our dependencies do not currently support VisionOS yet (for example, many of the Cocoapod libraries do not support it).

**Comment by StephenHodgson** [comment]
also depends on https://github.com/googlesamples/unity-jar-resolver/issues/628 to be completed first.

**Comment by mahdi862** [comment]
> We are looking into it, though it will take some time, as some of our dependencies do not currently support VisionOS yet (for example, many of the Cocoapod libraries do not support it).
Do you have a partially working repo that can be used? I only need the Auth and Realtime Database packages so it is very beneficial if the repo can be shared.

**Comment by StephenHodgson** [comment]
@a-maurice any updates?

**Comment by a-maurice** [comment]
Unfortunately nothing we can share at the moment, we've been busy trying to handle the upcoming privacy manifest and code signing changes for the Apple libraries.
On a positive, VisionOS support requires an ARM-based Mac, which GitHub actions has just recently started rolling out, which was one of the blockers preventing us from support.

**Comment by tom-haug** [comment]
@a-maurice Do you have even a rough timeframe on VisionOS support? I understand you are held up by "many of the Cocoapod libraries do not support it". Are you able to share which libraries these are? Are there alternatives being investigated to unblock progress?

**Comment by StephenHodgson** [comment]
@tom-haug one work around I was exploring was to manually add the visionOS platform capabilities to the iOS plugins in Unity, then in the exported xcode project trying to get firebase frameworks using swift package manager. No luck so far.

**Comment by a-maurice** [comment]
Unfortunately there isn't a timeline I can share at the moment.  There are generally two possible approaches, one being to get the Cocoapod solution to work, which requires some changes to other libraries and potentially Cocoapods itself; versus the other being to pivot to using Swift Package Manager, but the Swift / C++ interop layer is fairly new and changing, which hurts backwards compatibility.

And for the Firebase docs, the Firebase iOS SDK does support it, but only via Swift Package Manager, and not Cocoapods. Depending on which Firebase products you use, you could theoretically work directly with the iOS SDK instead of going through the Unity SDK, but that would very much be up to your individual use case.

**Comment by maxcelar** [comment]
Hi @a-maurice, thanks for elaborating on the issue back in April. Any news on the progress since then?

**Comment by argzdev** [comment]
Hi folks, thank you for taking an interest for this feature request. Please leave a thumbs up emoji on the author's original post so that we may track the demand of each feature/issue. That way, we can prioritize these features being added to our roadmap. Thanks!

**Comment by kevinsummers** [comment]
We have multiple enterprise opportunities lining up for this capability. @argzdev and @Rusus24 , I am encouraging our devs to add to the thumbs up counter ;) Thank you!

**Comment by chescored** [comment]
Firebase with Unity and Oculus Quest works fine until Firebase version 11.6.0, the following versions do not work since Firebase is compiling from 11.7.0 on Android SDK 33 and Oculus Quest at most to compile Android SDK 32. 

You may not notice any problems until you upload the application to the Meta board, where it does not allow uploading with the error message that the maximum Android SDK build version is 32.

Has anyone found a way to solve this? Can Firebase correct this problem in subsequent versions? Greetings.

**Comment by StephenHodgson** [comment]
@chescored your comment is off topic, but concerning enough to require its own thread. Please delete your comment and open a new issue about this directly. Thanks

**Comment by chescored** [comment]
Well, it still doesn't work, I have installed the Unity version 2022.3.38f but it still gives errors, I have changed the gradle files and there is no way.

Ergo Firebase only works with Oculus Quest up to version 11.6.0 as I mentioned.

**Comment by argzdev** [comment]
Hey @chescored, kindly refer back to the #1066 for future conversations.

**Comment by hackdamm** [comment]
Any news on the subject? i have been trying for days manually adjusting and adding everything but without sucess

**Comment by delphinius81** [comment]
Any updates here? My understanding is Cocoapods has been fully updated to support visionOS now.

**Comment by NikStires** [comment]
Any updates on this?

**Comment by Jake-NSW** [comment]
Hi is there any update on this? I've ended up having to reimplement the SDK using the rest API specifically only for visionOS and its fairly slow compared to the sdk. Thanks

**Comment by guerwan** [comment]
Hello, any update on this topic?
